src/paper_feeds/services/openalex.py

```diff
@@ -0,0 +1,51 @@
+import requests
+
+from paper_feeds.config import config
+
+OPENALEX_API_URL = 'https://api.openalex.org/'
+SOURCES_ENDPOINT = 'sources'
+USER_AGENT = 'paper-feeds (https://github.com/sneakers-the-rat/paper-feeds)'
+
+
+def _openalex_get(endpoint: str, params: dict) -> requests.Response:
+    # let OpenAlex know who you are, so you can enter their polite pool
+    # https://docs.openalex.org/how-to-use-the-api/rate-limits-and-authentication#authentication
+    if config.crossref_email:
+        params['mailto'] = config.crossref_email
+
+    headers = {'User-Agent': USER_AGENT}
+
+    # send out request to API
+    return requests.get(OPENALEX_API_URL + endpoint,
+                        params=params,
+                        headers=headers)
+
+
+def get_journal_homepages(issns: list[str]) -> dict:
+    """ Get homepage urls for a bunch of ISSNs. """
+
+    # minimize API calls using approach outlined in OpenAlex blog
+    # https://blog.ourresearch.org/fetch-multiple-dois-in-one-openalex-api-request/
+    slice_size = 50
```

Comment:
should this be hardcoded like this or any reason to make this a parameter, eg. if we want to chain this to some other updating call and control paging size all together?

Reply:
50 is the upper limit for chaining values via "OR" operator ([link](https://docs.openalex.org/how-to-use-the-api/get-lists-of-entities/filter-entity-lists#addition-or)).  
--> using slices of 50 entities reduces the amount of requests to the API
--> page size needs to be set accordingly, otherwise defaults to 25 and have to additionally paginate the resulting two pages